open_humans/member_views.py

```diff
@@ -97,12 +97,10 @@ def get_queryset(self):
             queryset = queryset.filter(project_members & authorized_members &
                                        visible_members & not_revoked)
 
-        projects = self.get_projects()
-        sorted_members = queryset.order_by('-id')
-        # Disabling for now until we can do this better
-#        sorted_members = sorted(queryset,
-#                                key=lambda m: projects.filter(member_id=m.id).count(),
-#                                reverse=True)
+        sorted_members = queryset.annotate(num_badges=Count('datarequestprojectmember__project',
+                                                            filter=Q(
+                                                                datarequestprojectmember__revoked=False
```

Comment:
Will this list people "higher" -- but display less badges for them if they have a lot of "hidden" project memberships? I'm wondering if it's better to just count visible memberships.

Reply:
good point